### PR TITLE
Add Cert Renewal information to Cert Management Guide

### PR DESCRIPTION
**Pull request by denyeart**
Add cert renewal and expiration guidance to the Cert Management Guide.

Signed-off-by: David Enyeart <enyeart@us.ibm.com>

### CODE REVIEW

File: docs/source/certs_management.md [hunk 1]
```diff
@@ -379,3 +379,77 @@ Certificate:
...
          eb:a8:97:1f:30:99:9d:ae:8d:ef:36:07:da:bb:67:ed:80
 ```
+
+## Certificate Renewal
+
+Enrollment certificates and TLS certificates for all nodes and clients should be updated before they expire in order to avoid disruption of service.
+Obtain updated certificates from the issuing Fabric Certificate Authority (CA) by using the Fabric CA enroll or reenroll function.
+The reenroll function allows the reuse of an existing private key, which is especially important for orderer TLS certificates.
+
+### Certificate Renewal of Expired Certificates
```
Comment (joshhus) [inline docs/source/certs_management.md]: Consider just "Renewal of Expired Certificates"

File: docs/source/certs_management.md [hunk 2]
```diff
@@ -379,3 +379,77 @@ Certificate:
...
+
+If certificates expire before reenrollment you must enable the `reenrollignorecertexpiry` option on the Fabric CA server prior to reenrollment.
```
Comment (joshhus) [inline docs/source/certs_management.md]: So this says user can "reenroll" either before or after expiration. Confirm that's true. Otherwise rephrase to make clearer.

File: docs/source/certs_management.md [hunk 3]
```diff
@@ -379,3 +379,77 @@ Certificate:
...
+
+Replace the peer's enrollment certificate or TLS certificate.
+
+The peer's enrollment certificate is configured to be in the peer's `<peer.mspConfigPath>/signcerts` directory (if private key was not reused the new private key should go to `<peer.mspConfigPath>/keystore` directory).
+
+The peer's TLS certificate is configured to be in the peer's `<peer.tls.cert.file>` directory/file (if private key was not reused the new private key should go to `<peer.tls.key.file>`).
+
+Restart the peer.
+
+### Renew orderer certificates
```
Comment (joshhus) [inline docs/source/certs_management.md]: initial caps for consistency w others. Renew Orderer Certificates. (that said, the trend is towards sentence style headings)

File: docs/source/certs_management.md [hunk 4]
```diff
@@ -379,3 +379,77 @@ Certificate:
...
+
+For orderer enrollment certificates you can also use the `enroll` or `reenroll` function, using the steps described above for peers.
+
+Orderer TLS certificates must be reenrolled with the option to reuse the private key. This is due to the orderer TLS certificates being configured in the system and application channels (orderer to orderer communication is verified using the public key in the configured TLS certificate). As of v1.4.9 or v2.2.1, orderer nodes verify a matching key rather than the entire configured TLS certificate, enabling orderer TLS certificate renewal without requiring a channel configuration update to update the certificate.
```
Comment (joshhus) [inline docs/source/certs_management.md]: system channels still supported, ok.

v1.4.9 and v2.2.1 ... I think, not one or the other.

File: docs/source/certs_management.md [hunk 5]
```diff
@@ -379,3 +379,77 @@ Certificate:
...
+
+Use `reenroll` and reuse the TLS certificate's private key by passing the `csr.keyrequest.reusekey` option and indicating the `mspdir` location of the existing private key (the private key is found in the <mspdir>/keystore directory):
```
Comment (joshhus) [inline docs/source/certs_management.md]: I don't think the build will publish the brackets right, it will think it's HTML. so use the ticks -  `<mspdir>/keystore`

File: docs/source/certs_management.md [hunk 6]
```diff
@@ -379,3 +379,77 @@ Certificate:
...
+
+```
+fabric-ca-client reenroll --csr.keyrequest.reusekey --mspdir <LOCATION-OF-IDENTITY'S-MSP-DIRECTORY-THAT-CONTAINS-THE-EXISTING-PRIVATE-KEY> --<OTHER-OPTIONS>`
+```
+
+Replace the orderer's enrollment certificate or TLS certificate.
+
+The orderer's enrollment certificate is configured to be in the orderer's `<General.LocalMSPDir>/signcerts` directory (if private key was not reused the new private key should go to `<General.LocalMSPDir>/keystore` directory).
```
Comment (joshhus) [inline docs/source/certs_management.md]: what does "should go to" mean exactly - maybe rephrase to be explicit. one prior instance above also.

File: docs/source/certs_management.md [hunk 7]
```diff
@@ -379,3 +379,77 @@ Certificate:
...
+
+The orderer's TLS certificate is configured to be in the orderer's `<General.TLS.Certificate>` directory/file (don't update the private key since it was reused).
+
+Restart the orderer node.
+
+### Renew Orderer TLS Certificates Without Reusing the Private Key
+
+While it is recommended to reuse the private key for orderer TLS certificate renewal, this may not be possible in all situations. Additional steps and planning are required since each of the ordering service channels must be updated for the new orderer TLS certificate.
```
Comment (joshhus) [inline docs/source/certs_management.md]: is the ordering service channels comment true for both system channel and non-system channel configs?

File: docs/source/certs_management.md (hunk 2 above)
Comment (denyeart) [reply docs/source/certs_management.md]: It's true, clarified!

File: docs/source/certs_management.md (hunk 7 above)
Comment (denyeart) [reply docs/source/certs_management.md]: true for both application and system channels, clarified!

File: docs/source/certs_management.md [hunk 8]
```diff
@@ -379,3 +379,76 @@ Certificate:
          9c:02:20:51:33:42:5e:a0:8a:2a:ec:f5:83:46:f0:99:6a:7e:
          eb:a8:97:1f:30:99:9d:ae:8d:ef:36:07:da:bb:67:ed:80
 ```
+
+## Certificate Renewal
+
+Enrollment certificates and TLS certificates for all nodes and clients should be renewed before they expire in order to avoid disruption of service.
```
Comment (C0rWin) [inline docs/source/certs_management.md]:
```suggestion
All nodes and clients should have their enrollment and TLS certificates renewed before expiring to keep the service from going down.
```

File: docs/source/certs_management.md [hunk 9]
```diff
@@ -379,3 +379,76 @@ Certificate:
...
+Obtain updated certificates from the issuing Fabric Certificate Authority (CA) by using the Fabric CA enroll or reenroll function.
```
Comment (C0rWin) [inline docs/source/certs_management.md]:
```suggestion
Use the Fabric CA enroll or reenroll function to get updated certificates from the issuing Fabric Certificate Authority (CA).
```

File: docs/source/certs_management.md [hunk 10]
```diff
@@ -379,3 +379,76 @@ Certificate:
...
+
+The orderer's enrollment certificate is configured to be in the orderer's `<General.LocalMSPDir>/signcerts` directory (if private key was not reused the new private key should also be updated in the `<General.LocalMSPDir>/keystore` directory).
+
+The orderer's TLS certificate is configured to be in the orderer's `<General.TLS.Certificate>` directory/file (don't update the private key since it was reused).
+
+Restart the orderer node.
+
+### Renew Orderer TLS Certificates Without Reusing the Private Key
+
+While it is recommended to reuse the private key for orderer TLS certificate renewal, this may not be possible in all situations. Additional steps and planning are required since each of the ordering service channels (including system channel, if used) must be updated for the new orderer TLS certificate.
```
Comment (C0rWin) [inline docs/source/certs_management.md]:
```suggestion
While it is recommended to reuse the private key for orderer TLS certificate renewal, this may not be possible in all situations. Additional steps and planning are required since each ordering service channel (including the system channel, if used) must be updated for the new orderer TLS certificate.
```

File: docs/source/certs_management.md [hunk 11]
```diff
@@ -379,3 +379,76 @@ Certificate:
...
+
+If you do not reuse the orderer TLS private key during reenrollment and the original orderer TLS certificates have not yet expired, you must update the orderer TLS certificates one at a time on each node and in each channel configuration and then verify orderer function before moving on to other orderer node TLS certificate updates.
```
Comment (C0rWin) [inline docs/source/certs_management.md]:
```suggestion
Suppose you do not reuse the orderer TLS private key during re-enrollment, and the original orderer TLS certificates have not yet expired. In that case, you must update the orderer TLS certificates one at a time on each node and in each channel configuration and then verify the orderer function before moving on to other orderer node TLS certificate updates.
```

**Review by C0rWin — COMMENTED** [review]
Suggested a few minor suggestions to improve the readability.

**Comment by denyeart** [comment]
Thanks @C0rWin , updates done, I think this one is ready for merge.

**Review by C0rWin — APPROVED** [review]
